### ISSUE TITLE
Feature Request: No property to access vertex neighbors in HalfEdgeTriangleMesh

### ISSUE THREAD

**Issue by MrGrayCode** [isl-org/Open3D#3154]
**Is your feature request related to a problem? Please describe.**
HalfEdgeTriangleMesh does not have any feature to compute the vertex neighbours. This leaves the tedious option of computing the 1-ring neighbours using the edge information.

**Describe the solution you'd like**
A class property to access vertex neighbours.

**Comment by benjaminum** [comment]
Would you be interested to contribute this feature? Please let me know, closing this issue until then.

**Comment by MrGrayCode** [comment]
@benjaminum Sure, I would like to take this up, but I am not sure how to contribute. Do I just add a function here : `cpp/open3d/geometry/HalfEdgeTriangleMesh.cpp` ?

**Comment by benjaminum** [comment]
@MrGrayCode Yes that is the right file. Python binding code will be in another file.
Can you provide a small code example (python and/or c++) of how the property would be used?
